### DISCUSSION TITLE
OutputFile equivalent in pester 5

### DISCUSSION THREAD

**Discussion by Agazoth** [pester/Pester#1940] [Q&A] [answered]
In Pester 4 you can publish an outputfile in Azure DevOps on the test pane of your pipeline run. Is there a similar way to achieve that in Pester 5?

**Comment by nohwnd** [comment, marked as answer]
How do you do that right now with Pester4? Asking to give you similar solution. 

In Pester 5 there is configuration object where you can enable output in NUnit and JUnit format. You would then use it with `Invoke-Pester -Configuration $c`
```
C:\p\pester [rel/5.2.x ≡]> $c = New-PesterConfiguration
C:\p\pester [rel/5.2.x ≡]> $c.TestResult

Enabled        : Enable TestResult. (False, default: False)
OutputFormat   : Format to use for test result report. Possible values: NUnitXml, NUnit2.5 or JUnitXml (NUnitXml,
                 default: NUnitXml)
OutputPath     : Path relative to the current directory where test result report is saved. (testResults.xml, default:
                 testResults.xml)
OutputEncoding : Encoding of the output file. (UTF8, default: UTF8)
TestSuiteName  : Set the name assigned to the root 'test-suite' element. (Pester, default: Pester)
```

Or maybe just using -CI switch would be enough for you. That enables the output to testResults.xml and exit on failure: 

https://github.com/pester/Pester/blob/8baf7344828e331aca8b09f0a536080911b14ec4/src/Main.ps1#L693-L695

**Reply by Agazoth** [reply]
I actually use a container to be able to use variable inpout to the pestertest and the -CI switch in the pipeline now.

To my understanding, Pester 5 only allows parameters in containers. Is there a way to combine parameter input and the xml outputfile? The Configuration object does not seem to allow Data as the Container does.

**Reply by nohwnd** [reply]
Yes, all the syntaxes translate to the config object internally. 

https://github.com/pester/Pester/blob/v5.0/src/Main.ps1#L719-L724

You would use Run.Container on the config object.

**Reply by nohwnd** [reply]
If you are using -CI already you should be getting testResults.xml already as well. You just need to upload it by using the publish results step. https://github.com/pester/Pester/blob/v5.0/azure-pipelines.yml#L133-L138

**Reply by Agazoth** [reply]
Thanks, @nohwnd - works like a charm!
